File: 1207. Unique Number of Occurrences/Solution.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int,int>f;
        for(auto ai:arr)f[ai]++;
        unordered_set<int>s;
        for(auto [k,v]:f){
            if(s.find(v)!=s.end())return 0;
            s.insert(v);
        }
        return 1;
    }
};

File: 1207. Unique Number of Occurrences/Solution.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> seen = new HashMap<>();
        Set<Integer> seenValues = new HashSet<>();
        for (int a : arr) {
            seen.put(a, seen.getOrDefault(a, 0) + 1);
        }
        for (Integer value : seen.values()) {
            if (seenValues.contains(value)) return false;
            seenValues.add(value);
        }
        return true;
    }
}
